### ISSUE TITLE
Incorrect chromosome length

### ISSUE THREAD

**Issue by noscode** [popsim-consortium/stdpopsim#701]
Hi,

**Short description of the problem:**
I tried to simulate and get vcf file for Homo Sapiens chromosome 1. And even the length of chromosome 1 in documentation is [248956422](https://stdpopsim.readthedocs.io/en/latest/catalog.html#sec_catalog_HomSap_genome) there were SNP's in vcf file  with greater positions (e. g. the last SNP was on position 249250516).

I have checked and it looks like the length of chromosome is taken from recombination map and it is equal to 249250621.

I have not find information about it, have I missed it?

**How to reproduce the problem:**
```python
import stdpopsim
species = stdpopsim.get_species("HomSap")
contig = species.get_contig("chr1", length_multiplier=1)
print(contig)
print(contig.recombination_map.get_length())  # 249250621
model = stdpopsim.IsolationWithMigration(10000, 5000, 5000, 4000, 0, 0)
samples = model.get_samples(10, 10)
engine = stdpopsim.get_default_engine()
ts = engine.simulate(model, contig, samples)
with open("stdpopsim_out.vcf", "w") as vcf_file:
    ts.write_vcf(vcf_file, contig_id='chr1', ploidy=2)
```

**Suggested fixes:**
If it is correct behavior then it will be good to have some note. If not then I am not sure where is the problem: in recombination map or in simulation which takes length from map. I have not checked other chromosomes and species.

**Comment by jeromekelleher** [comment]
Well spotted @noscode... I suspect I know what's wrong. Can you print out the last few site positions in the tree sequence before conversion to VCF please (.i.e., ``ts.tables.sites.position[-5:]``)?

**Comment by noscode** [comment]
Yes, sure:

```
[2.49228583e+08 2.49228865e+08 2.49235466e+08 2.49245907e+08
 2.49250034e+08]
```

**Comment by jeromekelleher** [comment]
Huh, that's weird, looks like the actual coordinates are spilling over. I thought this might be something to do with the VCF conversion. I'm going to mark this as a bug.

**Comment by noscode** [comment]
I have looked to the code and found that length is taken from the recombination map, but I did not do further digging for the reason why recombination map has wrong length.

**Comment by grahamgower** [comment]
Ah! Well the recombination maps usually stop before the end of the chromosome because recombination rates cannot be inferred there. So we should definitely be taking the contig length from the genome details for the species.

**Comment by grahamgower** [comment]
I tried to fix this just now, by extending the recombination map to the end of the chromosome. But it turns out that the hg37 hapmap recombination maps extend beyond the length of the b38 reference sequence we're now using in stdpopsim HEAD. What a pain!

**Comment by petrelharp** [comment]
I think this was closed by #720?

**Comment by grahamgower** [comment]
Sort of. It would more properly be closed when we have all the lifted-over recombination maps which match the chromosome length.

**Comment by nspope** [comment]
Closed by #1345
